File: backend-python/src/v1/models/user_entity.py
from sqlalchemy import Column, Integer, String, ForeignKey, Sequence
from sqlalchemy.orm import relationship
from src.db.base_class import Base


class User(Base):
    id = Column(Integer, Sequence('user_id_seq'), primary_key=True)
    name = Column(String(50), nullable=False)
    email = Column(String(50), nullable=False, unique=True)

    # Relationship to UserAddress
    addresses = relationship("UserAddress", back_populates="user", lazy="dynamic", cascade="all, delete-orphan")


class UserAddress(Base):
    id = Column(Integer, Sequence('user_address_id_seq'), primary_key=True)
    street = Column(String(50), nullable=False)
    city = Column(String(50), nullable=False)
    state = Column(String(50), nullable=False)
    zip_code = Column(String(10), nullable=False)

    # Relationship to User
    user_id = Column(Integer, ForeignKey('users.id'), nullable=False)
    user = relationship("User", back_populates="addresses")

File: backend-go/src/v1/entities/user-entity.go
package entities

func (UserEntity) TableName() string {
	return "go_api.users"
}
func (UserAddressEntity) TableName() string {
	return "go_api.user_addresses"
}

type UserEntity struct {
	Id        uint                `gorm:"primaryKey;autoIncrement"`
	Name      string              `gorm:"not null"`
	Email     string              `gorm:"not null;unique"`
	Addresses []UserAddressEntity `gorm:"foreignKey:UserID"`
}

type UserAddressEntity struct {
	Id      uint   `gorm:"primaryKey;autoIncrement"`
	Street  string `gorm:"not null"`
	City    string `gorm:"not null"`
	State   string `gorm:"not null"`
	ZipCode string `gorm:"not null"`
	UserID  uint   `gorm:"not null"`
}
